bblocks/import_tools/debt/wb_ids.py

```diff
@@ -0,0 +1,213 @@
+        return {
+            k: v["detailed_category" if detailed_category else "broad_category"]
+            for k, v in read_indicators("debt_service").items()
+        }
+
+    @classmethod
+    def debt_stocks_indicators(cls, detailed_category: bool = True) -> dict:
+        """Get a dictionary of Debt Service indicators in the IDS database."""
+        return {
+            k: v["detailed_category" if detailed_category else "broad_category"]
+            for k, v in read_indicators("debt_stocks").items()
+        }
+
+    def _get_indicator(
+        self, indicator: str, start_year: int, end_year: int
+    ) -> ImportData:
+        """Get data for an indicator. This method is not meant to be accessed
+        directly. Instead, use the `.get_data()` method.
+
+        Args:
+            indicator: The indicator to get. They must be in the IDS format
+                (e.g. DT.DOD.DECT.CD). To view all available indicators, call
+                `.get_available_indicators()`.
+
+        Returns:
+            The same object to allow chaining of methods
+        """
+
+        # Get a dataframe of the data
+        data = get_indicator_data(
+            indicator=indicator, start_year=start_year, end_year=end_year
+        )
+
+        # Rename the columns and convert the year column to a datetime
+        data = data.rename(
+            columns={"counterpart-area": "counterpart_area", "time": "year"}
+        ).astype({"year": "datetime64[ns]"})
+
+        # Save the data to a feather file
+        data.to_feather(self._path / f"{indicator}_{start_year}-{end_year}.feather")
+
+        return self
+
+    def load_data(
+        self, indicators: str | list, start_year: int, end_year: int
+    ) -> ImportData:
+        """Load the data for an indicator or a list of indicators.
+
+        Args:
+            indicators: The indicator(s) to load. They must be in the IDS format
+                (e.g. DT.DOD.DECT.CD). To view all available indicators, call
+                `.get_available_indicators()`.
+            start_year: The first year to include in the data
+            end_year: The last year to include in the data
+
+        """
+
+        # check if indicators is a string and convert to a list.
+        if isinstance(indicators, str):
+            indicators = [indicators]
+
+        # check if the indicators are valid
+        for indicator in indicators:
+            if indicator not in self.get_available_indicators().keys():
+                raise ValueError(f"{indicator} is not a valid indicator.")
+
+            # check if the data is already stored locally
+            stored_data = self._check_stored_data(
+                indicator, start_year=start_year, end_year=end_year
+            )
+
+            # if the data isn't stored locally, get it
+            if not stored_data:
+                # get the data
+                self._get_indicator(indicator, start_year, end_year)
```

Comment:
there is some inefficiency here, because if the indicator is already downloaded for different dates, some of the same data is extracted again

Reply:
Only if the requested data is not fully contained within the data already saved to disk. But then it's unavoidable to trigger a download, so may as well get the full period for which it was requested in order to avoid mixing different versions of the same data in a file.